build.gradle

```diff
@@ -71,6 +71,17 @@ embulkPlugin {
     mainClass = "org.embulk.input.ftp.FtpFileInputPlugin"
     category = "input"
     type = "ftp"
+
+    // "it.sauronsoftware:ftp4j:1.7.2" is not released in Maven Central, then,
+    // developers cannot build embulk-input-ftp with the depencency configuration.
+    //
+    // But, pom.xml is used also for a Maven-based plugin installation and loading.
+    // If an Embulk user specifies an appropriate Maven repository (ex. their own)
+    // which has "it.sauronsoftware:ftp4j:1.7.2" when installing embulk-input-ftp,
+    // they can use the Embulk plugin embulk-input-ftp.
+    additionalDependencyDeclarations = [
```

Comment:
when building jar and uploading to maven, is it still getting from `libs` folder? Is this setting using for running plugin only?

Reply:
No. Maven-formed archives do not include dependency libraries. Dependency librarires are only declared in `pom.xml`.

In summary,
* For compiling `embulk-input-ftp`, `libs/ftp4j-1.7.2.jar` is used just for references of classes and methods.
* For building a Maven-formed archive, `libs/ftp4j-1.7.2.jar` is NOT used. `pom.xml` expects that the user has their own Maven repository that has `it.sauronsoftware:ftp4j:1.7.2`.
* For building a RubyGem-formed archive, `libs/ftp4j-1.7.2.jar` is used.

Reply:
The `pom.xml` file is generated by gradle plugin, Is it possible to use local lib in `pom.xml` similar with build.gradle?

Reply:
Sorry, what do you mean? `pom.xml` will not have "local lib". Its `pom.xml` will have `it.sauronsoftware:ftp4j:1.7.2` as specified here.

Reply:
i mean,  are we able to use this https://stackoverflow.com/a/22300875/4022390? (not sure it is deprecated)

Because with the current gradle plugin, when we upgrade or add a library that is not in the maven repository, we need to
- update jar file in libs folder and build.gradle
- update jar version `additionalDependencyDeclarations` property in build.gradle
- upload jar file local maven repository
- manual config local maven repository in build.gradle then deploy & publish embulk plugin jar

Reply:
I know the current situation is not very good, but there is no good way to fix it while we use `ftp4j`.

The only good approach to deal with it would be: #29

Reply:
got it, thank you